crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,90 @@ impl Ellipse {
     }
 }
 
+/// An annulus primitive
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct Annulus {
+    /// The inner radius of the annulus
+    pub inner_radius: f32,
+    /// The outer radius of the annulus
+    pub outer_radius: f32,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_radius: 0.5,
+            outer_radius: 1.0,
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer radius
+    #[inline(always)]
+    pub const fn new(inner_radius: f32, outer_radius: f32) -> Self {
+        Self {
+            inner_radius,
+            outer_radius,
+        }
+    }
+
+    /// Get the diameter of the annulus
+    #[inline(always)]
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.outer_radius
+    }
+
+    /// Get the width of the annulus
+    #[inline(always)]
+    pub fn width(&self) -> f32 {
+        self.outer_radius - self.inner_radius
+    }
+
+    /// Get the area of the annulus
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        PI * (self.outer_radius.powi(2) - self.inner_radius.powi(2))
+    }
+
+    /// Get the perimeter or circumference of the annulus
+    #[inline(always)]
+    #[doc(alias = "circumference")]
+    pub fn perimeter(&self) -> f32 {
+        2.0 * PI * (self.outer_radius + self.inner_radius)
+    }
+
+    /// Finds the point on the annulus that is closest to the given `point`.
+    ///
+    /// If the point is outside the annulus, the returned point will be on
```

Comment:
As a reader, there are three cases I care about, not two.

1. Outside the annulus completely.
2. Overlapping with the annulus.
3. Inside the annulus's inner diameter.

Reply:
I agree, though now the doc seems a little verbose, should I cut something out?